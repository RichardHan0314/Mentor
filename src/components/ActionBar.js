import React from 'react';
import styled from 'styled-components';
import MoviesFilter from '../containers/MoviesFilter';
import MoviesSorting from '../containers/MoviesSorting';

const ActionBar = () => {
    return (
        <ActionBarContainer>
            <MoviesFilter />
            <MoviesSorting />
        </ActionBarContainer>
    );
};

const ActionBarContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: 10px 0;
    padding: 3px 0;
    border-bottom: 2px solid #555;
    box-shadow: 0 5px 5px -5px rgba(0, 0, 0, .5);
`;

export default ActionBar;
