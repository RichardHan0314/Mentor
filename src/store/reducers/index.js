/* eslint-disable import/prefer-default-export */
import { combineReducers } from 'redux';
import { moviesReducer } from './moviesReducer';
import { searchReducer } from './searchReducer';
import { filtersReducer } from './filtersReducer';
import { sortingReducer } from './sortingReducer';

export const rootReducer = combineReducers({
    movies: moviesReducer,
    search: searchReducer,
    filters: filtersReducer,
    sorting: sortingReducer
});
