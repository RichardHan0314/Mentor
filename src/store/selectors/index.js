/* eslint-disable import/prefer-default-export */
import { createSelector } from 'reselect';

export const getAllMovies = state => state.movies.movies;
export const searchValueSelector = state => state.search.searchValue;
export const getFilters = state => state.filters.filters;
export const getFiltersSelector = state => state.filters;
export const sortValueSelector = state => state.sorting.sortingValue;

// export const getFilter = (state) => {
//     // const { allFilters } = state.filters;
//     // const filterData = [];

//     // allFilters.filter(item => {
//     //     if (item.status === true) {
//     //         filterData.push(item.name);
//     //     }
//     //     return filterData;
//     // });

//     // return filterData;

//     const filters = getAllFilters(state);

//     return filters.reduce((filters, { status, name }) => {
//         return status ? [...filters, name] : filters;
//     }, [])
// };

// https://react-redux.js.org/api/hooks#equality-comparisons-and-updates
export const getActiveFilters = createSelector(getFilters, filters => {
    return Object.keys(filters).filter(key => filters[key]);
})

export const getParams = createSelector(
    [searchValueSelector, sortValueSelector, getActiveFilters],
    (search, sortBy, filter) => {
        return { search, searchBy: 'title', sortBy, filter: filter.join() }
    }
);