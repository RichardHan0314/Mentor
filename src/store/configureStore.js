/* eslint-disable import/prefer-default-export */
import { createStore, applyMiddleware, compose } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { rootReducer} from '../reducers';
// import { ping } from './enhancers/ping'; // <-- подключаем наш enhancer

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(rootReducer, composeEnhancers(applyMiddleware(thunk, logger)));