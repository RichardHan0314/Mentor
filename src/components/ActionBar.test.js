import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import ActionBar from './ActionBar';
import { store } from '../store/configureStore';

describe('ActionBar component', () => {
    test('render ActionBar snapshot', () => {
        const component = shallow(
            <Provider store={store}>
                <ActionBar />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
});