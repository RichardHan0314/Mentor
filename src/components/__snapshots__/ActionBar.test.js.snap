// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar component render ActionBar snapshot 1`] = `ShallowWrapper {}`;
